include/deal.II/grid/grid_tools.h

```diff
@@ -468,20 +468,29 @@ namespace GridTools
   /*@{*/
 
   /**
-   * Find and return the number of the used vertex in a given mesh that is
-   * located closest to a given point.
+   * Find and return the number of the used vertex (or marked vertex) in a
+   * given mesh that is located closest to a given point.
    *
    * @param mesh A variable of a type that satisfies the requirements of the
    * @ref ConceptMeshType "MeshType concept".
    * @param p The point for which we want to find the closest vertex.
+   * @param marked_vertices An array of bools indicating which
+   * vertices of @p mesh will be considered within the search
+   * as the potentially closest vertex. On receiving a non-empty
+   * @p marked_vertices, the function will
+   * only search among @p marked_vertices for the closest vertex.
+   * The size of this array should be equal to n_vertices() of the
+   * triangulation (as opposed to n_used_vertices() ).
```

Comment:
Write this as "...should be equal to the value returned by Triangulation::n_vertices() for the triangulation underlying the given mesh (as opposed to the value returned by Triangulation::n_used_vertices())." This way, doxygen will automatically link to these functions.